src/plugins/mmapstorage/mmapstorage.c

```diff
@@ -0,0 +1,1323 @@
+ * @param mmapHeader buffer where the MmapHeader is stored
+ * @param mmapMetaData buffer where the MmapMetaData is stored
+ *
+ * @retval -1 if fread() failed, magic number or format version was wrong
+ * @retval 0 if read succeeded, the magic number was read correctly and the format version matched
+ */
+static int readHeader (FILE * fp, MmapHeader * mmapHeader, MmapMetaData * mmapMetaData)
+{
+	memset (mmapHeader, 0, SIZEOF_MMAPHEADER);
+	memset (mmapMetaData, 0, SIZEOF_MMAPMETADATA);
+
+	if (fread (mmapHeader, SIZEOF_MMAPHEADER, 1, fp) != 1 || fread (mmapMetaData, SIZEOF_MMAPMETADATA, 1, fp) != 1)
+	{
+		return -1;
+	}
+
+	if (mmapHeader->mmapMagicNumber == ELEKTRA_MAGIC_MMAP_NUMBER && mmapHeader->formatVersion == ELEKTRA_MMAP_FORMAT_VERSION)
+	{
+		return 0;
+	}
+
+	return -1;
+}
+
+/**
+ * @brief Reads the MmapFooter from the end of the mapped region.
+ *
+ * @param mappedRegion pointer to the mapped region
+ * @param mmapHeader the MmapHeader containing the size of the allocation
+ *
+ * @retval -1 if the magic number did not match
+ * @retval 0 if the magic number matched
+ */
+static int readFooter (char * mappedRegion, MmapHeader * mmapHeader)
+{
+	MmapFooter * mmapFooter = (MmapFooter *) (mappedRegion + mmapHeader->allocSize - SIZEOF_MMAPFOOTER);
+
+	if (mmapFooter->mmapMagicNumber == ELEKTRA_MAGIC_MMAP_NUMBER)
+	{
+		return 0;
+	}
+
+	return -1;
+}
+
+/**
+ * @brief Writes the magic data for consistency checks.
+ *
+ * @param dest to write the data to.
+ */
+static void writeMagicData (char * dest)
+{
+	KeySet * destKeySet = (KeySet *) (dest + SIZEOF_MMAPHEADER + SIZEOF_MMAPMETADATA);
+	Key * keyPtr = (Key *) ((char *) destKeySet + SIZEOF_KEYSET);
+	memcpy (destKeySet, &magicKeySet, SIZEOF_KEYSET);
+	memcpy (keyPtr, &magicKey, SIZEOF_KEY);
+}
+
+/* -- Verification Functions  ----------------------------------------------------------------------------------------------------------- */
+
+/**
+ * @brief Generate magic number depending on pointer size.
+ *
+ * Generates magic number to detect arbitrary byte swaps.
+ * For each byte of the number, a different value is assigned.
+ * E.g.: Systems with 8-byte pointers will have the magic number 0x0706050403020100.
+ *
+ * @return generated magic number
+ */
+static uintptr_t generateMagicNumber (void)
+{
+	uintptr_t ret = 0;
+
+	size_t ptrBytes = sizeof (void *);
+	for (uintptr_t i = 0; i < ptrBytes; ++i)
+	{
+		ret |= (i << (i * 8));
+	}
+
+	return ret;
+}
+
+/**
+ * @brief Magic KeySet initializer.
+ *
+ * @param magicNumber to detect arbitrary byte-swaps
+ */
+static void initMagicKeySet (const uintptr_t magicNumber)
+{
+	magicKeySet.array = (Key **) magicNumber;
+	magicKeySet.size = SIZE_MAX;
+	magicKeySet.alloc = 0;
+	magicKeySet.cursor = (Key *) ~magicNumber;
+	magicKeySet.current = SIZE_MAX / 2;
+	magicKeySet.flags = KS_FLAG_MMAP_ARRAY | KS_FLAG_SYNC;
+	magicKeySet.mmapMetaData = (MmapMetaData *) 0xE1EF;
+#ifdef ELEKTRA_ENABLE_OPTIMIZATIONS
+	magicKeySet.opmphm = (Opmphm *) 0xFFFE;
```

Comment:
avoid magic constants like this (should be const int or macro)

Reply:
Done.